code/queue.rb

```diff
@@ -398,16 +398,11 @@ def run_queue_script!(msg)
           $log.debug("set ENV, now executing #{script_path}")
 
           # bash -lc will execute the command but first re-initializing like a new login (reading .bashrc, etc.)
-          exec_prefix = @config.exec_prefix || "bash -lc "
-          if exec_prefix.empty?
-            $log.debug("exec path: #{script_path}")
-            exec(script_path, "") if RUBY_VERSION < '2.0'
-            exec(script_path, "", :close_others => false)
-          else
-            $log.debug("exec path: #{exec_prefix + script_path}")
-            exec(exec_prefix + script_path) if RUBY_VERSION < '2.0'
-            exec(exec_prefix + script_path, :close_others => false)
-          end
+          exec_prefix = @config.exec_prefix.to_s.empty? ? "bash -lc" : @config.exec_prefix
```

Comment:
This makes it impossible for the config file to provide an empty exec_prefix. That's probably a reasonable change (in other words - my prior comment which says "well you could configure it this way, so the code is written to respect the possible values in JSON" would simply be replaced by "you can't configure it this way, an empty or null `exec_prefix` is considered invalid and will be replaced by the default value").